go/obscuronode/node.go

```diff
@@ -122,7 +121,9 @@ func sendInterrupt(interrupt *int32) *int32 {
 func (a *Node) processBlocks(blocks []common.EncodedBlock, interrupt *int32) {
 	var result enclave.SubmitBlockResponse
 	for _, block := range blocks {
-		result = a.Enclave.SubmitBlock(block.DecodeBlock().ToExtBlock())
+		if block != nil {
```

Comment:
Just out of curiosity, why would these blocks ever be nil ?

Reply:
good point, I should add a comment.
it's somewhat ugly. On the genesis block , the parent is null